app/src/main/java/com/example/playlistmaker/AudioPlayerActivity.kt

```diff
@@ -62,9 +86,78 @@ class AudioPlayerActivity : AppCompatActivity() {
         year.text = chosenTrack.releaseDate.take(4)
         genre.text = chosenTrack.primaryGenreName
         country.text = chosenTrack.country
-        songTime.text = SimpleDateFormat("mm:ss", Locale.getDefault()).format(chosenTrack.trackTimeMillis)
+        url = chosenTrack.previewUrl
+
+
+        play.setOnClickListener {
+            playbackControl()
+        }
 
+        preparePlayer()
+    }
+    override fun onPause() {
+        super.onPause()
+        pausePlayer()
+        if(isPlayerUsed){
+            handler.removeCallbacks(timePassedRunnable)
+        }
+    }
+    override fun onDestroy() {
+        super.onDestroy()
+        mediaPlayer.release()
+        if(isPlayerUsed){
+            handler.removeCallbacks(timePassedRunnable)
+        }
+    }
+    private fun preparePlayer() {
+        mediaPlayer.setDataSource(url)
+        mediaPlayer.prepareAsync()
+        mediaPlayer.setOnPreparedListener {
+            play.isEnabled = true
+            playerState = STATE_PREPARED
+            if(isPlayerUsed){
+                handler.removeCallbacks(timePassedRunnable)
+            }
+            songTime.text = getString(R.string.time)
+        }
+        mediaPlayer.setOnCompletionListener {
+            playerState = STATE_PREPARED
+            play.setImageDrawable(ResourcesCompat.getDrawable(resources, R.drawable.play, theme))
```

Comment:
```suggestion
            play.setImageDrawable(ContextCompat.getDrawable(this@AudioPlayerActivity, R.drawable.play))
```
Тема автоматически в этом случае достанется из контекста и тебе не нужно будет прокидывать её